tests/python/pants_test/engine/legacy/test_query_integration.py

```diff
@@ -0,0 +1,29 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest, ensure_daemon
+
+
+class ListQueryIntegrationTest(PantsRunIntegrationTest):
+  """Test the --query option, which is intended to subsume multiple other options and goals.
+
+  If v2 rules are made to satisfy both existing options such as --owner-of and their equivalent in
+  the --query option, the testing for that functionality should be moved into this file (or into
+  unit testing), and the separate option or goal should eventually be deprecated.
+  """
+  # TODO(#7346): test changes-since or changes-for-diffspec queries! It's currently difficult to set
+  # up an integration test for changing files.
+
+  @ensure_daemon
+  def test_query_success(self):
+    pants_run = self.do_command(
+      '--query=owner-of testprojects/tests/python/pants/dummies/test_pass.py',
+      'list',
+      success=True)
+    self.assertEqual(
+      pants_run.stdout_data.strip(),
```

Comment:
I'd rather assert that it equals exactly
`'testprojects/tests/python/pants/dummies:passing_target\n'`

In v1, pants has historically had a bunch of weird issues where it appends extra newlines to the end of output, and having testing to avoid that in v2 would make me pretty happy :)

Reply:
Let's absolutely do that! This was copied from the `--owner-of` integration test.

Reply:
Done, but the test still fails for strange reasons: `E   	Exception message: No installed @rules can satisfy Select(BuildFileAddresses) for input Params((Any+ScmWrapper)).`, but apparently only in `@ensure_daemon` testing (not just with pantsd on, that works 100% of the time), and I haven't investigated enough to figure out quite why yet.